ratatool-diffy/src/main/scala/com/spotify/ratatool/diffy/BigDiffy.scala

```diff
@@ -380,23 +431,32 @@ object BigDiffy extends Command {
     }
   }
 
+  /** for easier running via sbt */
+  def main(cmdlineArgs: Array[String]): Unit = run(cmdlineArgs)
+
   /** Scio pipeline for BigDiffy. */
   def run(cmdlineArgs: Array[String]): Unit = {
     val (sc, args) = ContextAndArgs(cmdlineArgs)
 
-    val (mode, key, lhs, rhs, output, header, ignore, unordered) = {
+    val (inputMode, key, lhs, rhs, output, header, ignore, unordered, outputMode) = {
       try {
-        (args("mode"), args("key"), args("lhs"), args("rhs"), args("output"),
+        (args("input-mode"), args("key"), args("lhs"), args("rhs"), args("output"),
           args.boolean("with-header", false), args.list("ignore").toSet,
-          args.list("unordered").toSet)
+          args.list("unordered").toSet, args("output-mode"))
       } catch {
         case e: Throwable =>
           usage()
           throw e
       }
     }
 
-    val result = mode match {
+    val saveToBq: Boolean = outputMode match {
```

Comment:
I think I prefer to leave this as a str for 2 reasons. 1 to match the current avro/bq impl for input. 2 so that it's easier to see the effects later on the code. Can use a `case match` instead of `if` later to split it out

Reply:
I can pass it in as a string and parse it in saveStats -- this has the downside that it will still run the analysis & only fail at the very end when saving. The code I have now throws an exception before doing any work if the output mode is invalid, since this branch is early in the main function. What do you think is better? Could also do a validation before running and still pass around the string but I find that a bit harder to follow, personally.

Reply:
Enum value?